pkgs/development/libraries/jpeg-xl/default.nix

```diff
@@ -0,0 +1,69 @@
+{ stdenv, lib, fetchgit
+, cmake
+, giflib
+, gperftools
+, gtest
+, libjpeg
+, libpng
+, libwebp
+, openexr
+, pkgconfig
+, zlib
+}:
+
+stdenv.mkDerivation rec {
+
+  pname = "jpeg-xl";
+  version = "2020-10-29-8f2e0857";
+
+  src = fetchgit { # as of writing, `fetchFromGitLab` cannot fetch submodules
+    # Temporary workaround for https://gitlab.com/wg1/jpeg-xl/-/issues/57
+    url = "https://gitlab.com/nh21/jpeg-xl";
+    # url = "https://gitlab.com/wg1/jpeg-xl";
+    rev = "8f2e08572e686b8016e1efac0a5eb5bc7fca1b81";
+    sha256 = "0i58vr1jmawma4jck3v2ym76adksknj8jasnxvfj96zs2q5y025i";
+    # There are various submodules in `third_party/`.
+    fetchSubmodules = true;
+    # The `CMakeLists.txt` errors on absence of `.git` directories.
+    leaveDotGit = true;
+  };
+
+  nativeBuildInputs = [
+    cmake
+    gtest
+    pkgconfig
+  ];
+
+  # Functionality not currently provided by this package
+  # that the cmake build can apparently use:
+  #     OpenGL
+  #     GLUT
+  #     Doxygen
+
+  # As of writing, jpeg-xl does not point out all its dependencies
+  # conclusively in its README or otherwise; they can best be determined
+  # by checking the CMake output for "Could NOT find".
+  buildInputs = [
+    giflib
+    gperftools # provides `libtcmalloc`
+    libjpeg
+    libpng
+    libwebp
+    openexr
+    zlib
+  ];
+
+  cmakeFlags = [
+  ];
```

Comment:
Is this meant to be empty?

Reply:
I kept it empty for now just to make clear to whoever would like to step up for maintenance that this is a cmake bulid and that this is where options would go.

We might for example be interested in these: https://gitlab.com/wg1/jpeg-xl/-/blob/4d70bd58fbcb758b61446aba447fedd9177a00c9/CMakeLists.txt#L98-104

```cmake
set(JPEGXL_FORCE_SYSTEM_GTEST false CACHE BOOL
    "Force using system installed googletest (gtest/gmock) instead of third_party/googletest source.")
set(JPEGXL_FORCE_SYSTEM_BROTLI false CACHE BOOL
    "Force using system installed brotli instead of third_party/brotli source.")
set(JPEGXL_FORCE_SYSTEM_HWY false CACHE BOOL
    "Force using system installed highway (libhwy-dev) instead of third_party/highway source.")
```

if we want it to use our system libs (but I'm also OK with it using its submodule libs, as I'm not sure if jpeg-xl often patches them).

Reply:
I think we should use system libs if they are available or add a comment to use them in the future except if they are heavily patched.

Reply:
Fine by me, but then we first need to have a good look into whether they changed important stuff or not.

Reply:
I've added a comment.